cmd/ocm-backplane/cloud/credentials.go

```diff
@@ -15,6 +16,13 @@ import (
 	"github.com/openshift/backplane-cli/pkg/utils"
 )
 
+var GetCloudCredential = getCloudCredential
+var GetCloudProvider = getCloudProvider
+var GetBackplaneClusterFromConfig = utils.GetBackplaneClusterFromConfig
+var Unmarshal = json.Unmarshal
+var PrintLn = fmt.Println
+var GetBackplaneConfiguration = config.GetBackplaneConfiguration
```

Comment:
Is it not an anti-pattern of assigning the variable to a function /closure? And it may lose readability too. 
If it needs, then it should be in a common module where everyone can reuse the same vars.

Reply:
That's a good point @samanthajayasinghe - do you know any other good ways to isolate methods so that they can be mocked/tested in unit tests? This felt like the cleanest option out of a bunch of quite ugly ones.

Reply:
cc @supreeth7 @Tafhim

Reply:
My thinking is to add a bunch of standard test methods for `config.GetBackplaneConfiguration` like function and setup via beforeEach()  and clean via afterEach(). 

`utils.GetBackplaneClusterFromConfig` also can be mocked via the temp kubeconfig file and return a fake cluster. 

`getCloudCredential` / `getCloudProvider` already mocked via `ClientUtilsMock` and possible to return fakeResponse or error.

Reply:
I agree, function assignment isn't a good approach while they might make the code look more structured.

@hectorakemp this overall looks great, can you try mocking `GetBackplaneClusterFromConfig` and make the necessary function calls (`GetBackplaneConfiguration`) before each test case?